SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -115,14 +149,41 @@ SpatialEventTracer::SpatialEventTracer(const FString& WorkerId)
 	}
 
 	FolderPath = EventTracePath;
-	const FString FullFileName = FString::Printf(TEXT("EventTrace_%s_%s.trace"), *WorkerId, *FDateTime::Now().ToString());
-	const FString FilePath = FPaths::Combine(FolderPath, FullFileName);
-
 	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	const FString FileName = TEXT("event-log");
+	const FString FileExt = TEXT(".etlog");
 	if (PlatformFile.CreateDirectoryTree(*FolderPath))
 	{
-		UE_LOG(LogSpatialEventTracer, Log, TEXT("Capturing trace to %s."), *FilePath);
-		Stream.Reset(Io_CreateFileStream(TCHAR_TO_ANSI(*FilePath), Io_OpenMode::IO_OPEN_MODE_WRITE));
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("Capturing trace file%s to %s."),
+			   (Settings->EventTraceFileOutputType == EEventTraceFileOutputType::Single) ? "" : "s", *FolderPath);
+
+		switch (Settings->EventTraceFileOutputType)
+		{
+		default:
+		case EEventTraceFileOutputType::Single:
+		{
+			const FString FullFilename = FString::Printf(TEXT("%s-%s%s"), *FileName, *WorkerId, *FileExt);
+			const FString FullFilePath = FPaths::Combine(FolderPath, FullFilename);
+			Stream.Reset(Io_CreateFileStream(TCHAR_TO_ANSI(*FullFilePath), Io_OpenMode::IO_OPEN_MODE_WRITE));
+			break;
+		}
+		case EEventTraceFileOutputType::Rotating:
+		{
+			const FString FullFilePathPrefix = FPaths::Combine(FolderPath, FileName);
+			const FString FullFilePathSuffix = WorkerId + FileExt;
```

Comment:
Oh! You have actually done pretty much exactly what I had in mind with these :P Nice!

One thing you will probably want to do if/when https://github.com/improbable/everything/pull/6044 is merged is manually add a `-` or `_` to the `FileName` because that will no longer added for you.

There is also then a debate about whether we make the worker ID part of the prefix or the suffix. The two options:
1.  include worker ID in the prefix, specifically such that the path ends up being `<saved_directory>/<worker_id>/event-log-<timestamp>.etlog` - this way, users can effectively specify a worker ID constraint by passing `<saved_directory>/<worker_id>/` as the "prefix" to search under. We can then support wildcards to get all of those e.g. `<saved_directory>/*/`.
2. include worker ID as in the suffix (i.e. no `<worker_id>` directory) - this way, the prefix to search under would in both cases be `<saved_directory>`, but when we make the GCS list API call, we would get GCS URIs for a bunch of event logs for all the workers under `<saved_directory>` (not sure how many this is as I think NVP divides this stuff up by pod and/or container). We may then have to provide a separate argument to the binary tool which can filter by worker ID (i.e. substring match the GCS URIs) locally so we don't fetch all of them.

So I guess it's remote vs. local filtering of URIs to download. Local filtering has the disadvantage that the number of list API call results potentially scales linearly with number of workers, and probably requires one more CLI parameter to be supported. The remote filtering option may also result in a slightly nicer UX when manually navigating the GCS buckets (because the ET logs will be grouped into folder).